src/android.ts

```diff
@@ -672,38 +701,32 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
                 return;
             }
 
-            await this._internal.requireRoot('enabling a WireGuard tunnel');
-
-            const { stdout: appUser } = await adb(['shell', 'stat', '-c', '%U', '/data/data/com.wireguard.android']);
-            await adb([
-                'shell',
-                'su',
-                appUser,
-                '/bin/sh',
-                '-c',
-                `"echo -n '${config}' > /data/data/com.wireguard.android/files/${tunnelName}.conf"`,
+            const { stdout: appUser } = await adbRootShell(['stat', '-c', '%U', '/data/data/com.wireguard.android']);
+            await adbRootShell([
+                `'su ${appUser} /bin/sh -c "echo -n '${Buffer.from(config, 'utf-8').toString(
+                    'base64'
+                )}' | base64 -d > /data/data/com.wireguard.android/files/${tunnelName}.conf"'`,
             ]);
 
             // We need to restart the WireGuard app for it to recognize our new tunnel config.
             await this.stopApp('com.wireguard.android');
 
-            await adb([
-                'shell',
+            await adbRootShell([
                 'am',
                 'broadcast',
                 '-a',
                 'com.wireguard.android.action.SET_TUNNEL_UP',
                 '-n',
                 // The quotes are necessary, otherwise `adb shell` interprets `$IntentReceiver` as a variable.
-                "'com.wireguard.android/.model.TunnelManager$IntentReceiver'",
+                "'com.wireguard.android/.model.TunnelManager\\$\\IntentReceiver'",
                 '-e',
                 'tunnel',
                 tunnelName,
             ]);
 
             const vpnIsStarted = await retryCondition(() => this._internal.isVpnEnabled());
             if (!vpnIsStarted) {
-                await deleteConfig();
+                // await deleteConfig();
```

Comment:
y?

Reply:
oops, that was leftover form development.